com.unity.render-pipelines.universal/ShaderLibrary/Lighting.hlsl

```diff
@@ -393,6 +397,17 @@ half3 DirectBDRF(BRDFData brdfData, half3 normalWS, half3 lightDirectionWS, half
 //                      Global Illumination                                  //
 ///////////////////////////////////////////////////////////////////////////////
 
+// Samples the Ambient Occlusion texture
+half SampleScreenSpaceOcclusionTexture(half3 positionCS)
+{
+#if defined(_SCREEN_SPACE_OCCLUSION)
```

Comment:
adding define to library function has some issues, if a user creates a custom shader he has to remember to declare this define. We had similar issues with SHADOW defines and people wanting to sample that from custom shaders.

I wonder if we should move the define check to not library (call site)? Thoughts?

Reply:
That makes sense. Good that you pointed that out. Prevents future issues from users (Y)